### PR TITLE
Rename zoltanParams to partitioningParams and adapt the metis parameter map function

### PR DESCRIPTION
**Pull request by lisajulia**
Since we have METIS support now, the kind of that parameter-map now depend on which partitioning option is chosen.

### CODE REVIEW

**Review by blattms — APPROVED** [review]

File: opm/grid/common/MetisPartition.cpp [hunk 1]
```diff
@@ -63,19 +62,63 @@ void setMetisOptions(const std::map<std::string, std::string>& optionsMap, idx_t
...
+            auto optionIt = metisOptionValues.find(value);
+            if (optionIt != metisOptionValues.end()) {
+                options[optionIndex] = optionIt->second;
+                Opm::OpmLog::info("Set metis option " + key + " to " + value + ".");
+            } else {
+                try {
+                    options[optionIndex] = std::stoi(value);
+                    Opm::OpmLog::info("Set metis option " + key + " to " + value + ".");
+                } catch (...) {
+                    std::cerr << "The value " << value << " for key " << key << " is not a valid METIS option." << std::endl;
```
Comment (blattms) [inline opm/grid/common/MetisPartition.cpp:108]: Maybe we should either warn or throw an exception please

File: opm/grid/common/MetisPartition.cpp [hunk 2]
```diff
@@ -63,19 +62,63 @@ void setMetisOptions(const std::map<std::string, std::string>& optionsMap, idx_t
...
+                }
+            }
+        } else if (std::any_of(metisOptionKeysForOtherMethods.begin(), metisOptionKeysForOtherMethods.end(), [&key](const std::string& otherKey) { return otherKey == key; })) {
+            std::cerr << "The METIS key " << key << " is not valid for METIS_PartGraphRecursive or METIS_PartGraphKway. Please choose only options for these graph partitioning functions." << std::endl;
```
Comment (blattms) [inline opm/grid/common/MetisPartition.cpp:112]: Either warn or throw an exception instead

File: opm/grid/common/MetisPartition.cpp [hunk 3]
```diff
@@ -63,19 +62,63 @@ void setMetisOptions(const std::map<std::string, std::string>& optionsMap, idx_t
...
+        } else if (key == "METIS_OPTION_PTYPE") {
+            manuallySelectedMethod = (value == "METIS_PTYPE_RB") ? 1 : 
+                                     (value == "METIS_PTYPE_KWAY") ? 2 :
+                                     3;
+            if (manuallySelectedMethod == 3) {
+                std::cerr << "Unknown value '" << value << "' for METIS option " << key << std::endl;
```
Comment (blattms) [inline opm/grid/common/MetisPartition.cpp:118]: dito

File: opm/grid/common/MetisPartition.cpp [hunk 4]
```diff
@@ -63,19 +62,63 @@ void setMetisOptions(const std::map<std::string, std::string>& optionsMap, idx_t
...
+            }
         } else {
-            std::cerr << "Unknown METIS option: " << key << std::endl;
+            std::cerr << "Unknown METIS key: " << key << std::endl;
```
Comment (blattms) [inline opm/grid/common/MetisPartition.cpp:121]: dito

**Review by blattms — COMMENTED** [review]
Looks good, I am just wondering whether we should really output to `std::cerr`

**Comment by lisajulia** [comment]
jenkins build this opm-simulators=5477 please

**Comment by blattms** [comment]
Just double checking:

**Comment by blattms** [comment]
jenkins build this opm-simulators=5477 serial please

**Pull request merged**